### ISSUE TITLE
Idiomatic regex filtering - feature or in documentation.

### ISSUE THREAD

**Issue by Ben-PH** [nushell/nushell#1051]
A very important part of productivity from the command line, is being able to filter using regex. For example, in bash, I might `ls \.*` to get all the hidden files/directories. I would like to see something like 
```
$  ls | regex <regex string>
```
In this case, column_list would default to just 
e.g. 
```
$ ls | regex '\.*' | pick name
━━━━┯━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
 #  │ name 
────┼───────────────────────────────────────
  0 │ .hidden_dir
  1 │ .hidden_file
━━━━┷━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
```
instead of
```
━━━━┯━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
 #  │ name 
────┼───────────────────────────────────────
  0 │ .hidden_dirfile
  1 │ .hidden_file
  2 | unhidden_file
━━━━┷━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
```

**Proposed solution** - assuming it's already in there and I've missed it
The Nu Book currently has a section on [strings](https://book.nushell.sh/en/types_of_data#strings). Add to that section how you can regex filter a table, targeting strings.

**Proposed solution** - not currently implemented
Add some sort of feature that allows you to control the filter with regex. Currently, you can do something like this:
```
$ ls | where name == ".hidden_file"
```
I would like to see something like these:
```
$ ls | where re == "<regular expression>"
$ ls | regex "..."
# this one would limit the regex to only work on the description colum, but still print all columns 
$ ls | regex --choose_col "<reg. ex.>" | pick description
```

**Describe alternatives you've considered**
I can't think of much in the way of alternatives. Maybe there's something that can be done by using native commands such as `grep`?

**Additional context**
In short, I would like to throw away table entries with more granulated control.

I have never used powershell, or any other shell with first-class objects, so I'm not so sure what the landscape of this problem is likely to be in reality.

**Comment by jonathandturner** [comment]
There are a couple of options available now:

You can use the `match` command:
```
ls | match name "^\."
```

For your particular example (I know  you meant something more general, but just for completeness):
```
ls .*
```

**Comment by forbesmyester** [comment]
Where is match? Is it in new versions? Is it in a branch?

```
/home/[username]> ls | match name '^\.'
error: Command not found
- shell:1:21
1 | ls | match name '^\.'
  |      ^^^^^ command not found
```

**Comment by jonathandturner** [comment]
`match` is a plugin. If you `cargo install`-ed Nu, you'll need to also `cargo install nu-plugin-match`.

If you're using it through the pre-built binaries, it should be shipped as part of it.

**Comment by martypapa** [comment]
For anyone reading, you may need to replace the hyphens with underscores:
`cargo install nu_plugin_match`
